rust/src/wilcoxon.rs

```diff
@@ -58,7 +67,8 @@ fn main() {
         Ok(_n) => {
             //println!("{} bytes read", n);
             //println!("input:{}", input);
-            const THRESHOLD: usize = 50; // Decrease this number so as to invoke the normal approximation for lower sample sizes. This would speed up the test at the cost of sacrificing accuracy.
+            const MIN_THRESHOLD: usize = 10; // Decrease this number so as to invoke the normal approximation for lower sample sizes. This would speed up the test at the cost of sacrificing accuracy.
```

Comment:
Why was this decreased from 50 to 10?

Reply:
This is because for calculating the exact p-value we need to consider all possible combinations of the test statistic. That is for vectors of length m and n, the total number of combinations will be (m+n)C(n) = (m+n)!/(m!n!) . In the example with both vectors of length 20 it will be 40!/(20!*20!) which is a very large number of combinations. That is why its hanging up.

Reply:
I am not sure how pnorm in R is calculating it. Its a bit opaque how R is calculating these values. There is not a lot of documentation, just how to use the syntax.